java/code/src/com/suse/manager/webui/websocket/VirtNotifications.java

```diff
@@ -133,8 +134,14 @@ public void onMessage(Session session, String messageBody) {
                 serverIds.add(newId);
 
                 // Send out a list of the latest pending actions to display
-                User user = (User)session.getUserProperties().get("currentUser");
-                Server server = SystemManager.lookupByIdAndUser(newId, user);
+                Optional<User> user = Optional.ofNullable(session.getUserProperties().get("webUserID"))
+                        .map(webUserID -> {
+                            User innerUser = UserFactory.lookupById((Long) webUserID);
+                            innerUser.getRoles();
+                            return innerUser;
+                        });
+
+                Server server = SystemManager.lookupByIdAndUser(newId, user.orElse(null));
```

Comment:
`lookupByIdAndUser` will fail with NPE if user is `null`. This was also failing in code prior this PR, but I'd take the chance to address this more cleanly if possible (in a separate commit)

Reply:
I protect the code to not reach the lookupByIdAndUser method is the user is null. 
I can protect that method, to not throw a NPE when user is null, and return a null object instead. But that method is used in 154 places, If I didn't protect where the method is used I will be only moving the place where the NPE appear. 
One good solution would be changing the method signature to return an optional. 
@moio Do you think I should invest time on this?

Reply:
> I can protect [lookupByIdAndUser] to not throw a NPE when user is null, and return a null object instead

Hmm, no, that would not be my suggestion. Rather, take as a fact that the method does not like nulls and avoid passing in nulls altogether.

Basically what I meant was that `user.orElse(null)` was not the right approach - if `user` is empty then the lookup should not be performed at all.

Reply:
I totally agree with you. I had change the code on friday to not perform the lookup if user is null. Thanks for the tips :)